extensions/wikia/ArticleVideo/bower_components/html5-skin/js/components/thumbnailCarousel.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * ThumbnailCarousel component
+ *
+ * @module ThumbnailCarousel
+ */
+
+var React = require('react'),
+    ReactDOM = require('react-dom'),
+    Utils = require('./utils');
+
+var ThumbnailCarousel = React.createClass({
+  getInitialState: function() {
+    return {
+      thumbnailWidth: 0,
+      thumbnailHeight: 0,
+      centerThumbnailWidth: 0,
+      centerThumbnailHeight: 0,
+      thumbnailPadding: 6
+    };
+  },
+
+  componentDidMount: function() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Function has a complexity of 15 which is greater than 10 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)